plugins/promela/hu.bme.mit.gamma.xsts.promela.transformation/src/hu/bme/mit/gamma/xsts/promela/transformation/serializer/ModelSerializer.xtend

```diff
@@ -205,9 +205,9 @@ class ModelSerializer {
 				«ENDFOR»
 				
 				«FOR index : 0 ..< actionSize»
-					chan_parallel_«index»?msg_parallel_«index»;
-					msg_parallel_«index» == 1;
-					msg_parallel_«index» = 0;
+					chan_parallel_«chanMapping.get(actions.get(index))»?msg_parallel_«chanMapping.get(actions.get(index))»;
```

Comment:
This `chanMapping.get(actions.get(index))` is called (duplicated) in five places. It would be worth extracting it into an auxiliary method.